kasa/smart/modules/colortemp.py

```diff
@@ -38,7 +40,7 @@ def query(self) -> dict:
     @property
     def valid_temperature_range(self) -> ColorTempRange:
         """Return valid color-temp range."""
-        return ColorTempRange(*self.data.get("color_temp_range"))
+        return ColorTempRange(*self.data.get("color_temp_range", DEFAULT_TEMP_RANGE))
```

Comment:
Perhaps it's better to perform `get` and log a warning on incrrect ranges? This way we know if we need to adjust this for some devices at some point.

Reply:
The range is missing from the hw version 1.0  so we need to set a default otherwise color_temp won't work. I've confirmed with @yaeuge that the default I got from the tapo app source is the actual min and max for his device.

Reply:
Yeah, but I mean that we could do something like this just in case for the highly unlikely case, that there are other devices with different characteristics. :
```
if (ct_range := self.data.get("color_temp_range") is None:
    _LOGGER.warning("Device doesn't report color temperature range, falling back to default %s", DEFAULT_TEMP_RANGE)
    ct_range = DEFAULT_TEMP_RANGE

return ColorTempRange(*ct_range)
```

Reply:
I can’t remember whether a library warning shows as default in HA? If not then it makes sense, otherwise it could spam logs for something we know happens.

Reply:
Yes, HA would show them. It can be a `_LOGGER.debug()` though, too. And I think HA should hopefully request this information only once.

Reply:
Done